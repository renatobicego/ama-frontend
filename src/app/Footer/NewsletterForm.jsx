"use client"
import { Input } from "@/app/utils/materialTailwind"
import { useState } from "react"
import validateEmail from "../utils/formValidation/emailValidation"


const NewsletterForm = ({socialMedia}) => {

    const [email, setEmail] = useState('')

    const handleEmailChange = e => setEmail(e.target.value)

    const handleSubmit = e => {
        e.preventDefault()
        console.log(validateEmail(email))
    }

    return(
        <form onSubmit={handleSubmit} className='flex flex-col justify-between gap-5 h-full w-full lg:w-1/3 parent-btn'>
            <h3 className='text-base md:text-xl font-semibold'>Suscribirse al Newsletter</h3>
            
            <Input 
                size="lg" 
                className="text-white" 
                color="gray" 
                placeholder='Ingrese su Mail*'
                value={email}
                onChange={handleEmailChange}
                />

            <div className='flex gap-8 justify-between items-center'>
                <button className='btn-primary' type="submit">Suscribirse</button>
                <nav>
                    <ul className='flex gap-2 items-center'>
                        {socialMedia.map((item, i) => {
                            return(
                                <a href={item.href} key={i}>
                                    <li><img src={item.logo} alt="" /></li>
                                </a>
                            )
                        })}
                    </ul>
                </nav>
            </div>
        </form>
    )
}

export default NewsletterForm