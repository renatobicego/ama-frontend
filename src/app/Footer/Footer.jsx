import Image from 'next/image'
import NewsletterForm from './NewsletterForm'
import Link from 'next/link'

const navFooterItems = [
    {
        text: 'Inscripción a Torneos',
        href: '/inscripciones/torneos'
    },
    {
        text: 'Federación AMA',
        href: '/inscripciones/federaciones'
    },
    {
        text: 'Resultados de Torneos',
        href: '/torneos'
    },
    {
        text: 'Noticias',
        href: '/noticias'
    },
    {
        text: 'Mi Perfil',
        href: '/perfil'
    },

]

const socialMedia = [
    // {
    //     href: '',
    //     logo: '/icons/Facebook.png'
    // },
    // {
    //     href: '',
    //     logo: '/icons/Twitter.png'
    // },
    {
        href: 'https://www.instagram.com/asociacionmendocinadeatletismo/',
        logo: '/icons/Instagram.png'
    }
]


const Footer = () => {
    return (
        <footer className='w-full bg-primary2 text-white border-t border-title py-14'>
            <div className='size-section flex gap-8 justify-between flex-wrap'>
                <section className='flex w-full lg:w-1/2 items-center justify-between flex-wrap gap-8'>
                    <div className='flex w-full gap-[8vw] lg:w-auto'>
                        <Image src="/icons/logoSinTitulo.png" alt="Logo AMA" width={70} height={70} className='max-w-[15vw] lg:max-w-[5vw]'/>
                        <div className='flex items-center gap-4 lg:hidden'>
                            <Image width={50} height={50} className='max-h-[12vw] lg:max-h-[3vw] w-auto' src='/icons/mail.svg' alt="" />
                            <div className='font-normal'>
                                <h4 className='text-sm md:text-base'>Email</h4>
                                <address className='text-sm md:text-base'>
                                    <Link href="mailto:mail@gmail.com">mail@gmail.com</Link>
                                </address>
                            </div>
                        </div>  
                    </div>
                    <nav className='w-full lg:w-1/2'>
                        <ul className='flex flex-col justify-between items-start gap-3 font-normal'>
                            {
                                navFooterItems.map((item, i) => {
                                    return(
                                        <a href={item.href} key={i} smooth="true">
                                            <li>{item.text}</li>
                                        </a>
                                )
                            })
                            }
                        </ul>
                    </nav>
                </section>
                
                <NewsletterForm socialMedia={socialMedia}/>
            </div>
           
        </footer>
    )
}

export default Footer