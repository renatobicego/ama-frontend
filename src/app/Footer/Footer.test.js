import { render, screen } from "@testing-library/react"
import Footer from "./Footer"
import '@testing-library/jest-dom/extend-expect'


describe('<Footer />', () => {

    let footer

    beforeEach(() => {
        footer = render(<Footer />)
    })

    test('renders content', () => {
        expect(footer.container).toMatchSnapshot();
        
        expect(footer.container).toHaveTextContent('Email')
        expect(footer.container).toHaveTextContent('Noticias')

        const logo = screen.getByAltText('Logo AMA')
        expect(logo).toBeInTheDocument()

        expect(footer.container).toMatchSnapshot();
        
    })
    
})