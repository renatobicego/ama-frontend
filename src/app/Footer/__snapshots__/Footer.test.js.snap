// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders content 1`] = `
<div>
  <footer
    class="w-full bg-primary2 text-white border-t border-title py-14"
  >
    <div
      class="size-section flex gap-8 justify-between flex-wrap"
    >
      <section
        class="flex w-full lg:w-1/2 items-center justify-between flex-wrap gap-8"
      >
        <div
          class="flex w-full gap-[8vw] lg:w-auto"
        >
          <img
            alt="Logo AMA"
            class="max-w-[15vw] lg:max-w-[5vw]"
            data-nimg="1"
            decoding="async"
            height="70"
            loading="lazy"
            src="/_next/image?url=%2Ficons%2FlogoSinTitulo.png&w=256&q=75"
            srcset="/_next/image?url=%2Ficons%2FlogoSinTitulo.png&w=96&q=75 1x, /_next/image?url=%2Ficons%2FlogoSinTitulo.png&w=256&q=75 2x"
            style="color: transparent;"
            width="70"
          />
          <div
            class="flex items-center gap-4 lg:hidden"
          >
            <img
              alt=""
              class="max-h-[12vw] lg:max-h-[3vw] w-auto"
              data-nimg="1"
              decoding="async"
              height="50"
              loading="lazy"
              src="/icons/mail.svg"
              style="color: transparent;"
              width="50"
            />
            <div
              class="font-normal"
            >
              <h4
                class="text-sm md:text-base"
              >
                Email
              </h4>
              <address
                class="text-sm md:text-base"
              >
                <a
                  href="mailto:mail@gmail.com"
                >
                  mail@gmail.com
                </a>
              </address>
            </div>
          </div>
        </div>
        <nav
          class="w-full lg:w-1/2"
        >
          <ul
            class="flex flex-col justify-between items-start gap-3 font-normal"
          >
            <a
              href="/inscripciones/torneos"
              smooth="true"
            >
              <li>
                Inscripci贸n a Torneos
              </li>
            </a>
            <a
              href="/inscripciones/federaciones"
              smooth="true"
            >
              <li>
                Federaci贸n AMA
              </li>
            </a>
            <a
              href="/torneos"
              smooth="true"
            >
              <li>
                Resultados de Torneos
              </li>
            </a>
            <a
              href="/noticias"
              smooth="true"
            >
              <li>
                Noticias
              </li>
            </a>
            <a
              href="/perfil"
              smooth="true"
            >
              <li>
                Mi Perfil
              </li>
            </a>
          </ul>
        </nav>
      </section>
      <form
        class="flex flex-col justify-between gap-5 h-full w-full lg:w-1/3 parent-btn"
      >
        <h3
          class="text-base md:text-xl font-semibold"
        >
          Suscribirse al Newsletter
        </h3>
        <div
          class="relative w-full min-w-[200px] h-11"
        >
          <input
            class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-500 text-white"
            placeholder="Ingrese su Mail*"
            value=""
          />
          <label
            class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-blue-gray-400 peer-focus:text-gray-500 before:border-blue-gray-200 peer-focus:before:!border-gray-500 after:border-blue-gray-200 peer-focus:after:!border-gray-500"
          >
             
          </label>
        </div>
        <div
          class="flex gap-8 justify-between items-center"
        >
          <button
            class="btn-primary"
            type="submit"
          >
            Suscribirse
          </button>
          <nav>
            <ul
              class="flex gap-2 items-center"
            >
              <a
                href=""
              >
                <li>
                  <img
                    alt=""
                    src="/icons/Facebook.png"
                  />
                </li>
              </a>
              <a
                href=""
              >
                <li>
                  <img
                    alt=""
                    src="/icons/Twitter.png"
                  />
                </li>
              </a>
              <a
                href=""
              >
                <li>
                  <img
                    alt=""
                    src="/icons/Instagram.png"
                  />
                </li>
              </a>
            </ul>
          </nav>
        </div>
      </form>
    </div>
  </footer>
</div>
`;

exports[`<Footer /> renders content 2`] = `
<div>
  <footer
    class="w-full bg-primary2 text-white border-t border-title py-14"
  >
    <div
      class="size-section flex gap-8 justify-between flex-wrap"
    >
      <section
        class="flex w-full lg:w-1/2 items-center justify-between flex-wrap gap-8"
      >
        <div
          class="flex w-full gap-[8vw] lg:w-auto"
        >
          <img
            alt="Logo AMA"
            class="max-w-[15vw] lg:max-w-[5vw]"
            data-nimg="1"
            decoding="async"
            height="70"
            loading="lazy"
            src="/_next/image?url=%2Ficons%2FlogoSinTitulo.png&w=256&q=75"
            srcset="/_next/image?url=%2Ficons%2FlogoSinTitulo.png&w=96&q=75 1x, /_next/image?url=%2Ficons%2FlogoSinTitulo.png&w=256&q=75 2x"
            style="color: transparent;"
            width="70"
          />
          <div
            class="flex items-center gap-4 lg:hidden"
          >
            <img
              alt=""
              class="max-h-[12vw] lg:max-h-[3vw] w-auto"
              data-nimg="1"
              decoding="async"
              height="50"
              loading="lazy"
              src="/icons/mail.svg"
              style="color: transparent;"
              width="50"
            />
            <div
              class="font-normal"
            >
              <h4
                class="text-sm md:text-base"
              >
                Email
              </h4>
              <address
                class="text-sm md:text-base"
              >
                <a
                  href="mailto:mail@gmail.com"
                >
                  mail@gmail.com
                </a>
              </address>
            </div>
          </div>
        </div>
        <nav
          class="w-full lg:w-1/2"
        >
          <ul
            class="flex flex-col justify-between items-start gap-3 font-normal"
          >
            <a
              href="/inscripciones/torneos"
              smooth="true"
            >
              <li>
                Inscripci贸n a Torneos
              </li>
            </a>
            <a
              href="/inscripciones/federaciones"
              smooth="true"
            >
              <li>
                Federaci贸n AMA
              </li>
            </a>
            <a
              href="/torneos"
              smooth="true"
            >
              <li>
                Resultados de Torneos
              </li>
            </a>
            <a
              href="/noticias"
              smooth="true"
            >
              <li>
                Noticias
              </li>
            </a>
            <a
              href="/perfil"
              smooth="true"
            >
              <li>
                Mi Perfil
              </li>
            </a>
          </ul>
        </nav>
      </section>
      <form
        class="flex flex-col justify-between gap-5 h-full w-full lg:w-1/3 parent-btn"
      >
        <h3
          class="text-base md:text-xl font-semibold"
        >
          Suscribirse al Newsletter
        </h3>
        <div
          class="relative w-full min-w-[200px] h-11"
        >
          <input
            class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-500 text-white"
            placeholder="Ingrese su Mail*"
            value=""
          />
          <label
            class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-blue-gray-400 peer-focus:text-gray-500 before:border-blue-gray-200 peer-focus:before:!border-gray-500 after:border-blue-gray-200 peer-focus:after:!border-gray-500"
          >
             
          </label>
        </div>
        <div
          class="flex gap-8 justify-between items-center"
        >
          <button
            class="btn-primary"
            type="submit"
          >
            Suscribirse
          </button>
          <nav>
            <ul
              class="flex gap-2 items-center"
            >
              <a
                href=""
              >
                <li>
                  <img
                    alt=""
                    src="/icons/Facebook.png"
                  />
                </li>
              </a>
              <a
                href=""
              >
                <li>
                  <img
                    alt=""
                    src="/icons/Twitter.png"
                  />
                </li>
              </a>
              <a
                href=""
              >
                <li>
                  <img
                    alt=""
                    src="/icons/Instagram.png"
                  />
                </li>
              </a>
            </ul>
          </nav>
        </div>
      </form>
    </div>
  </footer>
</div>
`;
